plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -1152,7 +1152,7 @@ def main():
                 module.exit_json(changed=False)
 
             proxmox_node = proxmox.nodes(vm[0]['node'])
-            if getattr(proxmox_node, VZ_TYPE)(vmid).status.current.get()['status'] == 'running':
+            if vm[0]['status'] == 'running':
                 module.exit_json(changed=False, msg="VM %s is running. Stop it before deletion." % vmid)
             taskid = getattr(proxmox_node, VZ_TYPE).delete(vmid)
```

Comment:
Ah, almost: this needs to be changed to
```suggestion
            taskid = proxmox_node.qemu.delete(vmid)
```

Reply:
Taken care of.